presto-thrift-connector-api/src/main/java/com/facebook/presto/connector/thrift/api/datatypes/PrestoThriftBigint.java

```diff
@@ -77,7 +78,7 @@ public Block toBlock(Type desiredType)
         int numberOfRecords = numberOfRecords();
         return new LongArrayBlock(
                 numberOfRecords,
-                nulls == null ? new boolean[numberOfRecords] : nulls,
+                Optional.ofNullable(nulls),
```

Comment:
The Thrift contract and original code seem a bit broken for the case of all nulls. If both the `nulls` and `longs` arrays are elided / null, I would expect that you get all nulls. However, the current code gives you `0` for every row. If we are fine clarifying the contract to cover this case, then we can use an RLE block.